yate: number regex routing fails

Maintainer: @jslachta
Environment: ar71xx, dir-825 b1, OpenWrt 18.06.1

Description:
I've been using regexroute.conf to do call routing in yate. I've had entries like this:
```
^12345$=fork ...
```
Then when a call for number 12345 is received it's forked to the phones I've specified.
This has worked well with lede 17.01.4, but stopped working when I upgraded to OpenWrt 18.06.1.

I've done some debugging and found that this:
```
^.*$=echo \0
```
.. prints "sip:12345", which seems to indicate that the "sip:" prefix is now included in the routing string.

When I ran lede 17.01.4 I compiled and used yate 6.0.0 successfully so I think that yate hasn't changed (OpenWrt 18.06.1 also includes yate 6.0.0). I've looked through the OpenWrt patches added the past year, but they don't look related to this problem.

So, I get a feeling that something in the yate environment has changed (musl or libraries?).

Has anyone noticed any similar behaviour?

Hello Robert,

I compiled yate 6.0.0 with the basic plugins on 17.01.6 and 18.06.0 and compared the output. I only saw one added warning related to sprintf use:

`message.cpp: In constructor 'TelEngine::MGCPMessage::MGCPMessage(TelEngine::MGCPEngine*, const char*, int, unsigned int, const char*, const char*)':
message.cpp:98:1: warning: '%03u' directive writing between 3 and 10 bytes into a region of size 4 [-Wformat-overflow=]
 MGCPMessage::MGCPMessage(MGCPEngine* engine, const char* name, int code,
 ^~~~~~~~~~~
message.cpp:98:1: note: directive argument in the range [0, 2147483647]
message.cpp:34:12: note: 'sprintf' output between 4 and 11 bytes into a destination of size 4
     sprintf(c,"%03u",code);
     ~~~~~~~^~~~~~~~~~~~~~~`

Seems unrelated.

I didn't expect to find anything, but thought I might as well just give it a try.

I think you should take this to yate support channels. Ask a la "I have input X, my config is like THIS, and I end up with this result. Is that to be expected?" Maybe ask on their mailing list. I think that would probably get more traction than asking here (probably same on OpenWrt forum).

I don't use yate myself, so can't say more.

Regards,
Seb

--- a/net/yate/Makefile
+++ b/net/yate/Makefile
@@ -10,7 +10,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=yate
 PKG_VERSION:=6.1.0-1
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://yate.null.ro/tarballs/yate6/
@@ -124,6 +124,10 @@ CONFIGURE_ARGS+= \
 	--without-coredumper \
 	--without-doxygen \
 	--without-kdoc
+
+# The regexp implementation of musl 1.1.19 is not fully compatible with yate
+CONFIGURE_ARGS+= \
+	--enable-internalregex
 
 ifneq ($(CONFIG_PACKAGE_$(PKG_NAME)-mod-isaccodec),)
 CONFIGURE_ARGS+=$(if $(CONFIG_SOFT_FLOAT),--disable-isac-float --enable-isac-fixed,--disable-isac-fixed --enable-isac-float)

